src/utils.ts

```diff
@@ -246,15 +246,47 @@ export function unixTime(): number {
   return Math.floor(Date.now() / 1000);
 }
 
-export const formatTimestamp = (t: number): string => {
-  return new Date(t * 1000).toLocaleString("en-EN", { dateStyle: "full", timeStyle: "long" });
+export const formatTimestamp = (timestamp: number, current = new Date().getTime()): string => {
+  const units: Record<string, number> = {
+    year: 24 * 60 * 60 * 1000 * 365,
+    month: 24 * 60 * 60 * 1000 * 365 / 12,
+    day: 24 * 60 * 60 * 1000,
+    hour: 60 * 60 * 1000,
+    minute: 60 * 1000,
+    second: 1000
+  };
+
+  // Multiplying timestamp with 1000 to convert from seconds to milliseconds
+  timestamp = timestamp * 1000;
+  const rtf = new Intl.RelativeTimeFormat('en', { numeric: "auto", style: 'long' });
+  const elapsed = current - timestamp;
+
+  if (elapsed > units["year"]) {
+    return new Date(timestamp).toUTCString(); // If it's more than a year we return early showing a Datetime string
+  } else if (elapsed < 0) {
+    return "now"; // If elapsed is a negative number we are dealing with an item from the future, and we return "now"
+  }
+
+  for (const u in units) {
+    if (elapsed > units[u] || u == 'second') {
+      // We convert the division result to a negative number to get "XX [unit] ago"
+      return rtf.format(Math.round(elapsed / units[u]) * -1, u as Intl.RelativeTimeFormatUnit);
+    }
+  }
+
+  throw Error("Not able to create relative Datetime string");
```

Comment:
Maybe instead of blowing up we just return `new Date(timestamp).toUTCString()`?

Reply:
According to the function this error is more like an `Unreachable` but yes the returning of a UTCString is a softer exit.